src/lib/utils/mergeDeep.ts

```diff
@@ -0,0 +1,29 @@
+import uniq from './uniq';
+import isPlainObject from './isPlainObject';
+
+/**
+ * Deeply merges all the object values in a new object.
+ *
+ * - Primitive values are replaced
+ * - Arrays a concatenated and their values are made unique
+ */
+function mergeDeep(...values: any[]): object {
```

Comment:
replace the any by:

```
type Item = {
  [key: string]: Array<Item> | Item
}
```

Something like that at least, will make it much clearer that this function only works for that

Reply:
eh, never mind, I forgot you can't make circular types in TS easily (and I forgot the end type), so any might be fine

Reply:
That wouldn't work in this line:

```js
mergeDeep(previousValue, nextValue)
```

We can use `object` instead of `any`. WDYT?

Reply:
yes, possible, it wasn't too clear what the types are to me, just based on this typing